### PR TITLE
calcOffset should always return a number

### PR DESCRIPTION
**Pull request by sunny-g**

### CODE REVIEW

**Comment by paranoidjk** [comment]
Have you came across any problem or find a bug?

Please give more detail info, thanks.

File: src/common/createSlider.jsx [hunk 1]
```diff
@@ -215,7 +215,7 @@ export default function createSlider(Component) {
     calcOffset(value) {
       const { min, max } = this.props;
       const ratio = (value - min) / (max - min);
-      return ratio * 100;
+      return isNaN(ratio) ? 0 : ratio * 100;
```
Comment (paranoidjk) [inline src/common/createSlider.jsx]: I don't konw in what case ratio could be `NaN`

- `max - min` should always more than `0` if uses give legal props

- `value - min` should always no less than `0`

So i think ratio should be always no less than `0` and no more than `1`

**Comment by benjycui** [comment]
Close for no response.

**Comment by sunny-g** [comment]
Sorry, can't remember the exact case in which this occurred, but it did happen and this PR fixed it. I believe the bug was that when the current `value` was out of bounds of the previously set `bounds`, `value` became `null`.

In either case, wouldn't you want some default implementation like this to avoid type conversion errors?

**Comment by benjycui** [comment]
No useless code, it will be confusing.

If you can re-produce, please provide an online demo, thanks.

Please provide a re-producible demo: http://codepen.io/benjycui/pen/aJooRE?editors=0011

**Pull request closed without merging**